modules/farm_fd2/src/entrypoints/cover_crop_seeding/lib.submitOptional.unit.cy.js

```diff
@@ -0,0 +1,128 @@
+import { lib } from './lib.js';
+
+describe('Submission using the cover_crop lib.', () => {
+  // form with no winter kill
+  let formNoWinterKill = {
+    date: '1950-01-02',
+    crops: ['BEAN', 'CARROT'],
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    areaSeeded: 50,
+    seedApplicationEquipment: ['Tractor'],
+    seedIncorporationEquipment: ['Rake'],
+    seedApplicationDepth: 6,
+    seedApplicationSpeed: 5,
+    seedIncorporationDepth: 8,
+    seedIncorporationSpeed: 3,
+    winterKill: false,
+    comment: 'A comment',
+  };
+
+  // form with no seed application equipment
+  let formNoSeedApplicationEquipment = {
+    date: '1950-01-02',
+    crops: ['BEAN', 'CARROT'],
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    areaSeeded: 50,
+    seedApplicationEquipment: [],
+    seedIncorporationEquipment: ['Rake'],
+    seedApplicationDepth: 0,
+    seedApplicationSpeed: 0,
+    seedIncorporationDepth: 8,
+    seedIncorporationSpeed: 3,
+    winterKill: true,
+    winterKillDate: '1950-12-31',
+    comment: 'A comment',
+  };
+
+  // form with no seed incorporation equipment
+  let formNoSeedIncorporationEquipment = {
+    date: '1950-01-02',
+    crops: ['BEAN', 'CARROT'],
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    areaSeeded: 50,
+    seedApplicationEquipment: ['Tractor'],
+    seedIncorporationEquipment: [],
+    seedApplicationDepth: 6,
+    seedApplicationSpeed: 5,
+    seedIncorporationDepth: 0,
+    seedIncorporationSpeed: 0,
+    winterKill: true,
+    winterKillDate: '1950-12-31',
+    comment: 'A comment',
+  };
+
+  before(() => {
```

Comment:
The `before` can be removed since it is no longer doing anything that isn't also done in the `beforeEach`